src/EnergyPlus/HeatBalFiniteDiffManager.cc

```diff
@@ -884,88 +894,119 @@ namespace HeatBalFiniteDiffManager {
-                                             SurfaceFD(SurfNum).condMaterialActuators(mat).isActuated,
-                                             SurfaceFD(SurfNum).condMaterialActuators(mat).actuatedValue);
+                                             SurfaceFD(SurfNum).condMaterialActuators(lay).isActuated,
+                                             SurfaceFD(SurfNum).condMaterialActuators(lay).actuatedValue);
                 EnergyPlus::SetupEMSActuator(state,
                                              "CondFD Surface Material Layer",
-                                             SurfaceFD(SurfNum).specHeatMaterialActuators(mat).actuatorName,
+                                             SurfaceFD(SurfNum).specHeatMaterialActuators(lay).actuatorName,
                                              "Specific Heat",
                                              "[J/kg-C]",
-                                             SurfaceFD(SurfNum).specHeatMaterialActuators(mat).isActuated,
-                                             SurfaceFD(SurfNum).specHeatMaterialActuators(mat).actuatedValue);
+                                             SurfaceFD(SurfNum).specHeatMaterialActuators(lay).isActuated,
+                                             SurfaceFD(SurfNum).specHeatMaterialActuators(lay).actuatedValue);
+            }
+
+            // Setup EMS Actuator and Output Variables for Heat Flux
+            // Only setup for layers 1 to N-1
+            for (int lay = 1; lay < state.dataConstruction->Construct(ConstrNum).TotLayers; ++lay) {
+                EnergyPlus::SetupEMSActuator(state,
+                                             "CondFD Surface Material Layer",
+                                             SurfaceFD(SurfNum).heatSourceFluxMaterialActuators(lay).actuatorName,
+                                             "Heat Flux",
+                                             "[W/m2]",
+                                             SurfaceFD(SurfNum).heatSourceFluxMaterialActuators(lay).isActuated,
+                                             SurfaceFD(SurfNum).heatSourceFluxMaterialActuators(lay).actuatedValue);
+                SetupOutputVariable(state,
+                                    format("CondFD Heat Source Power After Layer {}", lay),
+                                    OutputProcessor::Unit::W,
+                                    SurfaceFD(SurfNum).heatSourceFluxLayerReport(lay),
+                                    OutputProcessor::SOVTimeStepType::Zone,
+                                    OutputProcessor::SOVStoreType::State,
+                                    state.dataSurface->Surface(SurfNum).Name);
+                SetupOutputVariable(state,
+                                    format("CondFD Heat Source Energy After Layer {}", lay),
+                                    OutputProcessor::Unit::J,
+                                    SurfaceFD(SurfNum).heatSourceFluxEnergyLayerReport(lay),
```

Comment:
Looks like this is the total `QSSFlux` which may include energy from Construction:InternalSource along with the actuator energy. This is fine as an output variable, but shouldn't be on the electric meter. You'll need a separate output variable for the meter, like "CondFD EMS Actuator Heat Source Energy After Layer {}"?